src/index.ts

```diff
@@ -40,6 +52,40 @@ const createGenerator: CodegenGeneratorConstructor = (config, context) => {
 			return !!value.mediaType.mimeType.match('\\bjson$')
 		})
 
+		hbs.registerHelper('hasOperationSupportingMultipart', function(value: CodegenOperation | CodegenOperation[]): boolean {
+			if (isCodegenOperation(value)) {
+				return containsMultipartOperation([value])
+			} else {
+				return containsMultipartOperation(value)
+			}
+		})
+
+		hbs.registerHelper('toFileUploadProperties', function(properties: CodegenProperties): FileUploadProperty[] {
+			const results: FileUploadProperty[] = []
+			for (const prop in properties) {
+				const property = properties[prop]
+				if (property.schema.purpose !== CodegenSchemaPurpose.METADATA) {
+					continue
+				}
+
+				if (isCodegenArraySchema(property.schema) && property.schema.component) {
```

Comment:
I think we need to check here that `property.schema.component.schema.schemaType === CodegenSchemaType.FILE`.

Reply:
The correct `schemaType` is nested a bit deeper than that. We need to make a comparison against `property.schema.component.schema.properties.value.schema.schemaType`. However, this has a lot of ugly type-guards that need to be implemented.